File: client/aquila-ui/src/app/model/project.ts
import { UserResponse } from "./user";

export interface ProjectRequest {
    name: string,
    code: string,
    description: string,
    teamMembers: string[]
}

export interface ProjectResponse {
    readonly id: number,
    code: string,
    name: string,
    description: string,
    ticketCount: number,
    teamMembers: UserResponse[]
}

File: api/src/main/java/io/github/manishdait/aquila/project/Project.java
package io.github.manishdait.aquila.project;

import java.time.Instant;
import java.util.List;

import io.github.manishdait.aquila.users.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    @Column(unique = true)
    private String name;
    @Column(unique = true)
    private String code;
    private String description;
    private Instant createdAt;
    @ManyToMany
    private List<User> teamMembers;
}
